source/common/network/connection_impl.cc

```diff
@@ -593,7 +604,14 @@ void ConnectionImpl::onReadReady() {
   const bool latched_dispatch_buffered_data = dispatch_buffered_data_;
   dispatch_buffered_data_ = false;
 
-  ASSERT(!connecting_);
+  if (connecting_) {
```

Comment:
I think that internal sockets need to deliver a Write or Write|Read before just a Read event.  That way you can keep the original assert.

Reply:
Yeah, this block is not necessary. I will update

Reply:
I restore this block. It's another issue. I think I am following the always Write. Let's get back to connecting event handling